### PR TITLE
feat: remove support for down-leveling of TypeScript declarations files to ts3.9 to improve performance 

### PR DESCRIPTION
**Pull request by HBobertz**
BREAKING CHANGE: This PR removes the TypeScript declaration file down-leveling feature from the JSII compiler to address significant performance issues. Our benchmarks show that >50% of the JSII compile time is spent in the `downlevel-dts` dependency, causing JSII to be 2.7x slower than standard TypeScript compilation. The down-leveling feature was originally added to support TypeScript versions down to 3.9, but analysis shows minimal usage - none of our AWS CDK codebase uses this feature, and only about 12 non-fork repositories on GitHub would be affected by its removal, additionally typescript 3.9 and earlier have been EOL for quite a while on [definitively typed](https://github.com/DefinitelyTyped/DefinitelyTyped?tab=readme-ov-file#support-window). 

If you found this pr because removing this feature impacted your builds, it is likely because you have [this pattern](https://github.com/sandersn/downlevel-dts?tab=readme-ov-file#type-modifiers-on-importexport-names-45) somewhere in your codebase. Simply refactor that pattern and rebuild. Ensure you also are not seeing the below typesVersion in your app's package.json
```json
  "typesVersions": {
    "<=3.9": {
      "*": [
        ".types-compat/ts3.9/*",
        ".types-compat/ts3.9/*/index.d.ts"
      ]
    }
  },
```

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

**Comment by HBobertz** [comment]
Getting test failures I'm not getting locally. I'll figure that out

**Comment by mrgrain** [comment]
Let's probably not make this a `chore` but a `fix`.

File: fixtures/@scope/jsii-calc-base/lib/index.ts [hunk 1]
```diff
@@ -36,4 +36,4 @@ export class StaticConsumer {
   }
 }
 
-export * as deep from './deep';
+export * as deep from './deep';
```
Comment (mrgrain) [inline fixtures/@scope/jsii-calc-base/lib/index.ts:39]:
```suggestion
export * as deep from './deep';

```

File: src/compiler.ts [hunk 2]
```diff
@@ -21,6 +19,7 @@ import * as utils from './utils';
 const LOG = log4js.getLogger('jsii/compiler');
 export const DIAGNOSTICS = 'diagnostics';
 export const JSII_DIAGNOSTICS_CODE = 9999;
+export const TYPES_COMPAT = '.types-compat';
```
Comment (mrgrain) [inline src/compiler.ts:22]:
```suggestion
```

File: test/compiler.test.ts [hunk 3]
```diff
@@ -3,8 +3,7 @@ import { tmpdir } from 'node:os';
 import { dirname, join } from 'node:path';
 import { loadAssemblyFromPath, SPEC_FILE_NAME, SPEC_FILE_NAME_COMPRESSED } from '@jsii/spec';
 import { compile, Lock } from './fixtures';
-import { Compiler } from '../src/compiler';
-import { TYPES_COMPAT } from '../src/downlevel-dts';
+import { Compiler, TYPES_COMPAT } from '../src/compiler';
```
Comment (mrgrain) [inline test/compiler.test.ts:6]:
```suggestion
import { Compiler } from '../src/compiler';
```

Comment (mrgrain) [reply test/compiler.test.ts:6]: Please adjust the tests in this file (and possibly code)
`TYPES_COMPAT` should not be emitted anymore.

**Review by mrgrain — CHANGES_REQUESTED** [review]
small changes left

**Review by mrgrain — APPROVED** [review]

**Pull request merged**